homeassistant/components/fronius/config_flow.py

```diff
@@ -104,6 +115,46 @@ async def async_step_import(self, conf: dict) -> FlowResult:
         """Import a configuration from config.yaml."""
         return await self.async_step_user(user_input={CONF_HOST: conf[CONF_RESOURCE]})
 
+    async def async_step_dhcp(self, discovery_info: DhcpServiceInfo) -> FlowResult:
+        """Handle a flow initiated by the DHCP client."""
+        for entry in self._async_current_entries(include_ignore=False):
+            if entry.data[CONF_HOST].lstrip("http://").rstrip("/").lower() in (
+                discovery_info.ip,
+                discovery_info.hostname,
+            ):
+                return self.async_abort(reason="already_configured")
+        # Symo Datalogger devices need up to 1 minute at boot from DHCP request
+        # to respond to API requests (connection refused until then)
+        await asyncio.sleep(DHCP_REQUEST_DELAY)
+        try:
+            unique_id, self.info = await validate_host(self.hass, discovery_info.ip)
+        except Exception:  # pylint: disable=broad-except
```

Comment:
Usually we try to avoid broad exception handlers.

Can this be `CannotConnect`?

Reply:
I guess CannotConnect would suffice.

I was basically copying this schema from the user step - which was copied from the scaffold script https://github.com/home-assistant/core/blob/67f9118cbf38e08fdb615f88dcbeea80fc98bce6/script/scaffold/templates/config_flow/integration/config_flow.py#L92
Since this runs without any user interaction I think it makes sense to catch broad here. 
- `aiohttp.ClientError` 
- `aiohttp.ContentTypeError`
- `json.decoder.JSONDecodeError`
- `asyncio.TimeoutError` 
- and `KeyError` at unpacking 
are handled by `pyfronius` - not sure if there can raise something else, like `OSError`.

Reply:
should I except both, CannotConnect and Exception and do different abort reasons / log the exception if not CannotConnect?

Reply:
We usually only use the broad except in the `user` step.

Ideally we don't catch exceptions we haven't planned for because it will hide bugs and make them harder to track down. If the library is expected to throw `OSError` then we should catch that as well. 

I couldn't find another integration doing a broad except catch for `dhcp` (is possible I missed it since I'm a bit tired)
 `git grep -A 50 async_step_dhcp`

Reply:
No OSError isn't expected - we are just making http calls with aiohttp.
Most integrations using DHCP use the mac address as unique_id. This doesn't work with Fronius since the API doesn't respond it. So we need to make an API call in the DHCP step to get the unique_id. 
Let's change it to CannotConnect and see if at one point an issue is raised by a user - I wouldn't expect it.